### PR TITLE
Rename `SparseQArray` to `SparseDIAQArray`

### PR DESCRIPTION
**Pull request by pierreguilmin**

### CODE REVIEW

**Review by abocquet — CHANGES_REQUESTED** [review]
We are not planning to implement another format for now so there is no need to specify this I think

**Comment by gautierronan** [comment]
> We are not planning to implement another format for now so there is no need to specify this I think

Yes, but if we do implement another sparse format, it's easier to make this change now than to have to do it later on (might break the API).

File: dynamiqs/qarrays/sparse_dia_qarray.py [hunk 1]
```diff
@@ -298,16 +298,16 @@ def _construct_diags(offsets: tuple[int, ...], other: ArrayLike) -> Array:
     return diags
 
 
-def to_sparse(x: DenseQArray | Array) -> SparseQArray:
-    r"""Returns the input matrix in the `SparseQArray` format.
+def to_sparse_dia(x: DenseQArray | Array) -> SparseDIAQArray:
```
Comment (gautierronan) [inline dynamiqs/qarrays/sparse_dia_qarray.py:301]: I am conflicted on this API. WDYT about `def to_sparse(..., format=dq.SparseQArray) -> QArray` or something like that?

Comment (pierreguilmin) [reply dynamiqs/qarrays/sparse_dia_qarray.py:301]: Meh, why don't you like this version? The other sparse formats may need extra constructor arguments that we are not yet aware of.

**Pull request merged**